examples/calibrate_tsqrtP.py

```diff
@@ -0,0 +1,322 @@
+            beta_v_t = beta_v_t["t_pulse", decreasing_idx[-1] + 1 :]
+        # }}}
+        plt.axhline(  # the linear threshold is the threshold above which beta is linear
+            y=linear_threshold / 1e-6,
+            color=thiscolor,
+            label=f"linear threshold for amp={amplitude}",
+        )
+        # {{{ flip data so beta is on x axis now and t_pulse is the y axis
+        t_v_beta = (
+            beta_v_t.shape.alloc(dtype=np.float64)
+            .rename("t_pulse", r"$\beta$")
+            .setaxis(r"$\beta$", beta_v_t.data)
+        )
+        t_v_beta.data[:] = (
+            beta_v_t["t_pulse"].copy() / 1e-6
+        )  # because our ppg wants μs
+        t_v_beta.set_units("μs").set_units(r"$\beta$", "s√W")
+        # use as temp for ultimate coeff
+        # }}}
+        # {{{ Determine linear and nonliear coefficients
+        c_nonlinear = t_v_beta[
+            r"$\beta$":(None, linear_threshold)
+        ].C  # make copy to allow the next
+        #                                                               line not to cause the data to
+        #                                                               actually shift
+        c_nonlinear[
+            r"$\beta$"
+        ] -= linear_threshold  # Taylor expand around the linear threshold rather than 0
+        c_nonlinear = c_nonlinear.polyfit(r"$\beta$", order=10)
+        c_linear = t_v_beta[r"$\beta$":(linear_threshold, None)].polyfit(
+            r"$\beta$", order=1
+        )
+        print(
+            f"\n**************** Coefficients for {amplitude} ****************\n"
+        )
+        print("Non-linear regime coefficients:\n", c_nonlinear)
+        print("Linear regime coefficients:\n", c_linear)
+
+        # }}}
+        def prog_plen(desired):
+            """If the desired beta is above the linear threshold,
+            the linear fit is applied to return the pulse length
+            required to obtain the desired beta.
+            If the desired beta is below the linear threshold, the
+            nonlinear fit is applied to return the pulse length
+            required to obtain the desired beta.
+            If a series of desired betas is fed, both fits are
+            applied appropriately to return the pulse lengths
+            required to obtain each desired beta in the series.
+            """
+
+            def zonefit(desired):
+                if desired > linear_threshold:
+                    return np.polyval(c_linear[::-1], desired)
+                else:
+                    return np.polyval(
+                        c_nonlinear[::-1], desired - linear_threshold
+                    )
+
+            ret_val = np.vectorize(zonefit)(desired)
+            if ret_val.size > 1:
+                return ret_val
+            else:
+                return ret_val.item()
+
+        fl.next(r"Amplitude*$t_{pulse}$ vs $\beta$", legend=True)
+        t_v_beta.set_plot_color_next()
```

Comment:
since you are already using this, and it should be simpler -- can you use this instead of manually pulling the color cycler above?